src/tests/system/tests/test_ldap_extra_attrs.py

```diff
@@ -42,3 +48,196 @@ def test_ldap_extra_attrs__filled(client: Client, provider: GenericProvider, att
+    )
+
+    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
+    assert user_dict["description"] == ["gecos1"], "attribute 'descripion' was not correct"
+    assert user_dict["mail"] == ["user1@example.test"], "attribute 'mail' was not correct"
+    assert user_dict["email"] == ["user1@example.test"], "attribute 'email' was not correct"
+
+
+@pytest.mark.ticket(bz=1667252)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__ifp(client: Client, provider: GenericProvider):
+    """
+    :title: ifp do not crash when requesting extra attributes
+    :setup:
+        1. Create new user "user1"
+        2. Configure 'test' ldap user extra attribute
+        3. Start SSSD
+    :steps:
+        1. Run "sssctl user-checks user1"
+        2. Check SSSD status
+    :expectedresults:
+        1. Command succeeded
+        2. Checked successfully
+    :customerscenario: True
+    """
+    provider.user("user1").add()
+    client.sssd.sssd["services"] = "nss, pam, ifp"
+    client.sssd.domain["ldap_user_extra_attrs"] = "test:homeDirectory"
+    client.sssd.ifp["user_attributes"] = "+test"
+    client.sssd.start()
+
+    result = client.sssctl.user_checks("user1")
+    assert result.rc == 0, "sssctl user-checks command failed"
+
+    result = client.sssd.svc.status("sssd")
+    assert result.rc == 0, "service status sssd failed"
+
+
+@pytest.mark.ticket(bz=847043)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_ldap_extra_attrs__thread_issue(client: Client, provider: GenericProvider):
+    """
+    :title: Thread issue can cause the application to not get any identity information
+    :setup:
+        1. Create new user "user1"
+        2. Edit config - "use_fully_qualified_names" to "false"
+        3. Start SSSD
+    :steps:
+        1. Start new thread with provided function
+        2. Main thread waits for worker and then sleeps
+        3. Worker is in loop calling getpwuid()
+        4. Main thread cancel worker thread (with exit_flag.set())
+        5. Thread join
+        6. Call getpwuid()
+    :expectedresults:
+        1. Thread is started
+        2. Success
+        3. Success
+        4. Worker is cancelled
+        5. Thread is joined
+        6. Call is not stucked
+    :customerscenario: True
+    """
+
+    def worker():
+        nonlocal started, exit_flag
+        uid = os.geteuid()
+        started.set()
+        while not exit_flag.is_set():
+            pwd.getpwuid(uid)
+
+    provider.user("user1").add()
+    client.sssd.domain["use_fully_qualified_names"] = "false"
+    client.sssd.start()
+
+    started = threading.Event()
+    exit_flag = threading.Event()
+
+    thread = threading.Thread(target=worker)
+    thread.start()
+
+    started.wait()
+    time.sleep(3)
+
+    exit_flag.set()
+    thread.join()
+
+    pw = pwd.getpwuid(os.geteuid())
+    assert pw.pw_name, "pwd.getpwuid() call did not return anything"
```

Comment:
I don't think this tests anything.

* python threads are only simulated, they are not real threads
* all code here is executed on the localhost not on the client (both threads and getpwuid)